src/representations/representation_manager_spherical_expansion.hh

```diff
@@ -0,0 +1,535 @@
+                               "been implemented");
+        return -1;
+      }
+    };
+
+    /** Radially-dependent template specialization of the above */
+    template<>
+    struct AtomicSmearingSpecification<GaussianSigmaType::Radial> {
+      explicit AtomicSmearingSpecification(json ) {}
+      template<size_t Order, size_t Layer>
+      double get_gaussian_sigma(ClusterRefKey<Order, Layer> & /* pair */) {
+        throw std::logic_error("Requested a sigma type that has not yet "
+                               "been implemented");
+        return -1;
+      }
+    };
+
+
+
+  }
+
+  /**
+   * Handles the expansion of an environment in a spherical and radial basis.
+   *
+   * The local environment of each atom is represented by Gaussians of a
+   * certain width (user-defined; can be constant, species-dependent, or
+   * radially dependent).  This density field is expanded in an angular basis
+   * of spherical harmonics (à la SOAP) and a radial basis of either Gaussians
+   * (again, as in SOAP) or one of the more recent bases currently under
+   * development.
+   */
+
+  template<class StructureManager>
+  class RepresentationManagerSphericalExpansion:
+    public RepresentationManagerBase {
+   public:
+    using hypers_t = RepresentationManagerBase::hypers_t;
+    using Property_t = Property<double, 1, 1, Eigen::Dynamic, Eigen::Dynamic>;
+    using Manager_t = StructureManager;
+
+    /**
+     * Set the hyperparameters of this descriptor from a json object.
+     *
+     * @param hypers structure (usually parsed from json) containing the
+     *               options and hyperparameters
+     *
+     * @throw logic_error if an invalid option or combination of options is
+     *                    specified in the structure
+     */
+    void set_hyperparameters(const hypers_t& hypers) {
+      this->max_radial = hypers.at("max_radial");
+      this->max_angular = hypers.at("max_angular");
+      if (hypers.find("n_species") != hypers.end()) {
+        this->n_species = hypers.at("n_species");
+      } else {
+        this->n_species = 1; // default: no species distinction
+      }
+      this->interaction_cutoff = hypers.at("interaction_cutoff");
+      this->cutoff_smooth_width = hypers.at("cutoff_smooth_width");
+      this->gaussian_sigma_str =
+                hypers.at("gaussian_sigma_type").get<std::string>();
+      this->radial_ortho_matrix.resize(this->max_radial, this->max_radial);
+      this->radial_sigmas.resize(this->max_radial, 1);
+      this->radial_norm_factors.resize(this->max_radial, 1);
+      this->radial_nl_factors.resize(this->max_radial, this->max_angular + 1);
+      this->soap_vectors.resize_to_zero();
+      //this->soap_vectors.resize(this->n_species * this->max_radial,
+                                //pow(this->max_angular + 1, 2));
+      this->is_precomputed = false;
+
+      this->hypers = hypers;
+      if (this->gaussian_sigma_str.compare("Constant") == 0) {
```

Comment:
It isn't robust, it's even case sensitive (because there's no one-line way to do case-insensitive string comparison in standard C++, as far as I can tell). That said, why would there be leading or trailing spaces in a JSON param value? That would be quite obvious, as the value is enclosed in quotes.